internal/pkg/cli/deploy.go

```diff
@@ -244,31 +391,89 @@ func (o *deployOpts) Run() error {
 		return err
 	}
 
-	if err := o.maybeInitWkld(); err != nil {
+	deploymentOrderGroups, err := o.getDeploymentOrder()
+	if err != nil {
 		return err
 	}
 
-	if err := o.loadWkld(); err != nil {
-		return err
+	cmds := make([][]workloadCommand, len(deploymentOrderGroups))
+	// Do all our asking before executing deploy commands.
+	for i, deploymentGroup := range deploymentOrderGroups {
+		for _, workload := range deploymentGroup {
+			if err := o.maybeInitWkld(workload); err != nil {
+				return err
+			}
+			deployCmd, err := o.loadWkldCmd(workload)
+			if err != nil {
+				return err
+			}
+			cmds[i] = append(cmds[i], workloadCommand{
+				name: workload,
+				cmd:  deployCmd,
+			})
+			if err := deployCmd.Ask(); err != nil {
+				return fmt.Errorf("ask %s deploy: %w", o.wlType, err)
+			}
+			if err := deployCmd.Validate(); err != nil {
+				return fmt.Errorf("validate %s deploy: %w", o.wlType, err)
+			}
+		}
 	}
-	if err := o.deployWkld.Execute(); err != nil {
-		return fmt.Errorf("execute %s deploy: %w", o.wlType, err)
+
+	// Count number of deployments.
+	count := 0
+	for i := 0; i < len(cmds); i++ {
+		count += len(cmds[i])
 	}
-	if err := o.deployWkld.RecommendActions(); err != nil {
-		return err
+	log.Infof("Will deploy %d %s in the following order.\n", count, english.PluralWord(count, "workload", ""))
+	for i := 0; i < len(cmds); i++ {
+		names := ""
+		for _, cmd := range cmds[i] {
+			names += cmd.name + " "
+		}
+		log.Infof("%d. %s\n", i+1, names)
 	}
+
+	totalNumDeployed := 1
```

Comment:
I was initially confused of why we initialized this to 1, because from the variable named, it seemed to represent the number of workload _already_ deployed, which should be initialized `0` since nothing has been deployed yet.

For its usage at L443, I understood why it's initialized as 1.

What do you think instead of having `totalNumDeployed`, we have:
```
for i, deploymentGroup := range cmds {
    for g, cmd := range deploymentGroup {
       // ....i + g

```

Reply:
good idea; this is not actually doing what I want it to--that is, printing "deployment 1 of 4 in group 1".